From affb92de5c4a0e613a284e181e78dec0b558f7b8 Mon Sep 17 00:00:00 2001
From: Venkataramana Kotakonda <venkataramana.kotakonda@intel.com>
Date: Thu, 13 Nov 2014 12:15:24 +0530
Subject: [PATCH 8/8] CHT-CCSM: Fixed ACA RID resolve and detection issue in
 pmic ccsm driver

Description: When PMIC detect ACA charger, and if ccsm driver is not able
resolve the RID then driver is reporting it as SDP. But if RID is not
resolved by ccsm driver, then charger type should be reported as ACA as
PMIC detected it as ACA to allow charging.

Jira Id: https://jira01.devtools.intel.com/browse/AKP-607
Change-Id: I825fdef19b7d8b65702db1195edd12685785f2f2
Signed-off-by: Venkataramana Kotakonda <venkataramana.kotakonda@intel.com>
---
 drivers/power/intel_pmic_ccsm.c | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/drivers/power/intel_pmic_ccsm.c b/drivers/power/intel_pmic_ccsm.c
index 827c2a6..ad4deec 100644
--- a/drivers/power/intel_pmic_ccsm.c
+++ b/drivers/power/intel_pmic_ccsm.c
@@ -782,7 +782,7 @@ static int pmic_get_usbid(void)
 	u8 val;
 
 	ret = pmic_read_reg(chc.reg_map->pmic_schgrirq1, &val);
-	if (!ret)
+	if (ret)
 		return RID_UNKNOWN;
 
 	/* SCHGRIRQ1_REG SUSBIDDET bit definition:
@@ -803,7 +803,7 @@ static int pmic_get_usbid(void)
 		dev_err(chc.dev, "IIO channel read error for USBID\n");
 		goto err_exit;
 	}
-
+	dev_dbg(chc.dev, "%s: rid=%d\n", __func__, rid);
 	if (IS_RID_A(rid))
 		id = RID_A;
 	else if (IS_RID_B(rid))
@@ -857,7 +857,9 @@ static int get_charger_type(void)
 		rid = pmic_get_usbid();
 		if (rid == RID_A)
 			return POWER_SUPPLY_CHARGER_TYPE_ACA_DOCK;
-		else if (rid != RID_UNKNOWN)
+		/* As PMIC detected the charger as ACA, if RID detection
+		 * failed report type as ACA  */
+		else
 			return POWER_SUPPLY_CHARGER_TYPE_USB_ACA;
 	case PMIC_CHARGER_TYPE_SE1:
 		return POWER_SUPPLY_CHARGER_TYPE_SE1;
-- 
1.9.1

